### ISSUE TITLE
Bug: Major Flaw in conda

### ISSUE THREAD

**Issue by arita37** [conda/conda#6049]
When installing mkl with
```
 conda install mkl

The following NEW packages will be INSTALLED:
    boost-cpp:      1.65.1-vc9_0        conda-forge [vc9]
    icc_rt:         2017.0.4-h97af966_0 anaconda
    intel-openmp:   2018.0.0-hcd89f80_7 anaconda
    mkl:            2018.0.0-h36b65af_4 anaconda

The following packages will be REMOVED:

    cffi:           1.10.0-py27_0       defaults
    conda:          4.3.21-py27_0       defaults
    cryptography:   1.8.1-py27_0        defaults
    idna:           2.5-py27_0          defaults
    ipaddress:      1.0.16-py27_0       defaults
    menuinst:       1.4.1-py27_0        defaults
    openssl:        1.0.2h-vc9_0        defaults    [vc9]
    pip:            9.0.1-py27_1        defaults
    pycosat:        0.6.1-py27_1        defaults


```

but, Pycosat is fundamental module in conda
----> Then, conda is broken !!!!!!!!!!!!!!!!!!!!

Solution :  
   Could you make conda self-embed all dependencies (sub-package of conda),
   there is no risk of deleteting conda dependencies.... ?

  Could you make a .tar backup of current conda source (before any overwrite/update) ?
   At least, we can have a manual process to recover it...

**Comment by kalefranz** [comment]
What is the output of `conda info`?

**Comment by arita37** [comment]
conda should often more protected from bad update/dependencies issues...
 ---> self embed conda is the safest way to protect it.


(root) asus1@ASUS1-PC E:\
$ conda info
Current conda install:

```
               platform : win-64
          conda version : 4.3.24
       conda is private : False
      conda-env version : 4.3.24
    conda-build version : 1.21.14+0.g4dfebe9.dirty
         python version : 2.7.13.final.0
       requests version : 2.14.2
       root environment : D:\_devs\Python01\Anaconda27  (writable)
    default environment : D:\_devs\Python01\Anaconda27
       envs directories : D:\_devs\Python01\Anaconda27\envs
                          C:\Users\asus1\AppData\Local\conda\conda\envs
                          C:\Users\asus1\.conda\envs
          package cache : D:\_devs\Python01\Anaconda27\pkgs
                          C:\Users\asus1\AppData\Local\conda\conda\pkgs
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/anaconda/win-64
                          https://conda.anaconda.org/anaconda/noarch
                          https://conda.anaconda.org/r/win-64
                          https://conda.anaconda.org/r/noarch
                          https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : C:\Users\asus1\.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.24 requests/2.14.2 CPython/2.7.13 Windows/7 Windows/6.1.7601
          administrator : True
```

**Comment by kalefranz** [comment]
Would love to figure out how conda scheduled pycosat to be uninstalled here, but I unfortunately cannot reproduce it.

The functionality as you describe will be accomplished according to #3912.  It's a tricky problem, and implementation is moving forward, but still not yet complete.

**Comment by arita37** [comment]
Can you see carefull my 1st message ?

By installing, conda install mkl
conda asked to remove conda ....


Do you think it makes sense ?








On 6 Oct 2017, at 02:40, Kale Franz <notifications@github.com> wrote:

Closed #6049.

―
